### PR TITLE
Add name field to workshop form

### PR DESCRIPTION
**Pull request by TurnerRiley**
Adds a "Workshop Name" field to the workshop builder/editor form.

### Create a workshop
Doesn't show the name field for non-BYO workshops:
DOESNT SHOW FOR ALL WORKSHOPS

Shows the name field for BYO workshops:
SHOWS FOR BYOW

Can't submit the form without entering a name:
CANT SUBMIT WITHOUT NAME

Shows read-only name field after submitted:
Shows readonly of name after created

Name is shown in workshop database entry:
SQL SHOWS DATA ENTRY

### Edit a workshop
Can edit the name:
Can edit the name

Name is updated in workshop database entry:
Edited sql

## Links
Jira ticket: [here](https://codedotorg.atlassian.net/jira/software/c/projects/ACQ/boards/64?assignee=60d62161f65054006980bd71&selectedIssue=ACQ-2836)

## Testing story
Local testing and updating unit tests.

### CODE REVIEW

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx [hunk 1]
```diff
@@ -1106,6 +1120,26 @@ export class WorkshopForm extends React.Component {
...
+              <Col sm={4}>
+                <FormGroup validationState={validation.style.name}>
+                  <ControlLabel>Workshop Name</ControlLabel>
+                  <FormControl
+                    type="text"
+                    value={this.state.name || ''}
+                    id="name"
+                    name="name"
+                    onChange={this.handleFieldChange}
+                    maxLength={255}
```
Comment (bethanyaconnor) [inline apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:1134]: Is this just the default value for this prop? Would love to break this out so it’s not a magic number but that might be too big of a change 😬

File: apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js [hunk 2]
```diff
@@ -1,14 +1,12 @@
-import {fireEvent, render, screen} from '@testing-library/react';
+import {fireEvent, render, screen, waitFor} from '@testing-library/react';
+import '@testing-library/jest-dom';
 import React from 'react';
 import {Provider} from 'react-redux';
+import sinon from 'sinon'; // eslint-disable-line no-restricted-imports
```
Comment (bethanyaconnor) [inline apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js]: Why is sinon needed?

Comment (bethanyaconnor) [reply apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js]: Said another way -- using jest is preferred. Can jest be used or is there a reason sinon must be used here?

File: apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx (hunk 1 above)
Comment (TurnerRiley) [reply apps/src/code-studio/pd/workshop_dashboard/components/workshop_form.jsx:1134]: I saw that the [other text field components](https://github.com/code-dot-org/code-dot-org/pull/62841/files#diff-f0655af74c25a070d365fc92a5d25247d6f3bc0d24e209e3ea0cafdf9fe3f25cR1153) in this file had this same prop (I think to match the 255 character limit of the database entry). I could make it a file constant to clean it up a bit?

File: apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js (hunk 2 above)
Comment (TurnerRiley) [reply apps/test/unit/code-studio/pd/professional_learning_landing/landingPageTest.js]: Oh I can switch to jest!

**Review by bethanyaconnor — APPROVED** [review]

**Pull request merged**